Compiler/General.py

```diff
@@ -110,32 +114,41 @@ def get_set_cell_value_code(new_value, previous_value, zero_next_cell_if_necessa
     def get_abc(offset):
         # returns a,b,c such that a*b+c=offset and a+b+c is minimal
 
-        min_a, min_b, min_c = offset, 1, 0
-        min_sum = min_a + min_b + min_c
+        min_a, min_b, min_c = 0, 0, 0
+        min_sum = offset + 1
+
+        left = 1
+        right = offset // 2 - 1
 
-        for i in range(1, offset // 2 + 1):
-            a, b, c = i, offset // i, offset % i
-            curr_sum = a + b + c
+        while right >= left:
+            a, b = left+1, right+1
+            c = offset - a * b
+            curr_sum = abs(a) + abs(b) + abs(c)
```

Comment:
Can a,b be negative?
I think that adding negative values (c) only makes it more complicated :(

If you want to cover the case where there can be <a> * <b> plus signs, followed by <c> minus signs, then the previous way was covering this by decreasing <b> by 1 and keeping only plus signs (just an example)
(I thought about this algorithm a lot)

Reply:
Only b will be negative if offset is negative. And if offset is negative c is inverted

Reply:
a and b inside of get_abc can never be negative